### ISSUE TITLE
[ARM32/Linux] Cross architecture build error in PAL assembly codes

### ISSUE THREAD

**Issue by hseok-oh** [dotnet/coreclr#9818]
build command:
./build.sh release cross arm crosscomponent

Error message
```
[ 25%] Building ASM object src/pal/src/CMakeFiles/coreclrpal.dir/arch/arm/context2.S.o
/nfs/rbp3/coreclr/src/pal/src/arch/arm/context2.S:13:1: error: unknown directive
.syntax unified
^
/nfs/rbp3/coreclr/src/pal/src/arch/arm/context2.S:14:1: error: unknown directive
.thumb
^

<errors in assembly code>

make[2]: *** [src/pal/src/CMakeFiles/coreclrpal.dir/arch/arm/context2.S.o] Error 1
make[1]: *** [src/pal/src/CMakeFiles/coreclrpal.dir/all] Error 2
```

It seems to releated to the change of src/pal/src/CMakeLists.txt in #9650. It tried to compile asm file for arm, but it must have tried to compile asm for x86.

**Comment by gkhanna79** [comment]
@hseok-oh Can this be closed?

**Comment by hseok-oh** [comment]
Fixed by #9819 and remain issue #9902
